src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -53,40 +53,61 @@ public Assembly LoadFromPath(string fullPath)
 
         private Assembly LoadFromPathUnchecked(string fullPath)
         {
+            var assemblyIdentity = AssemblyIdentityUtils.TryGetAssemblyIdentity(fullPath);
+            return LoadFromPathUncheckedCore(fullPath, assemblyIdentity);
+        }
+
+        private Assembly LoadFromPathUncheckedCore(string fullPath, AssemblyIdentity identity)
```

Comment:
can it be LoadFromPathUnchecked(string, identity = null)? looks like there is only 2 caller of this method.

and do identity = identity ?? AssemblyIdentityUtils.TryGetAssemblyIdentity(fullpath);